### PR TITLE
Additional config fields added 

### PR DESCRIPTION
**Pull request by prachidamle**
Changes to support:
1) Additional fields to config.yml [maintainer, license, projectURL] 
2) readme.md file under each template, a link will be provided in the response
3) output section in .catalog section of templates

https://github.com/rancher/rancher/issues/2643

### CODE REVIEW

File: manager/catalog_manager.go [hunk 1]
```diff
@@ -66,13 +64,15 @@ func SetEnv() {
 //Init clones or pulls the catalog, starts background refresh thread
 func Init() {
 	_, err := os.Stat(catalogRoot)
-	if err == nil {
+	if os.IsExist(err) || err == nil {
```
Comment (alena1108) [inline manager/catalog_manager.go]: If os.stat returns path error, there would be no need to call the code block below. As the path error would indicate that the directory you are looking for, doesn't exist. Unless I'm missing something

Comment (prachidamle) [inline manager/catalog_manager.go]: This needs to be if !os.IsNotExist(err) || err == nil {..}

os.IsNotExist(err) will be true if the dir does not exists. We will just negate it to check if dir exists.

File: manager/catalog_manager.go [hunk 2]
```diff
@@ -3,6 +3,9 @@ package manager
 import (
 	"flag"
 	"fmt"
+	log "github.com/Sirupsen/logrus"
```
Comment (cloudnautique) [inline manager/catalog_manager.go]: Typically, you want to see these sorted like they were... the gopkg and github separated by a line from the builtin Golang libs.

**Comment by prachidamle** [comment]
@cloudnautique I update the PR after reordering the imports

File: manager/catalog_manager.go (hunk 2 above)
Comment (prachidamle) [inline manager/catalog_manager.go]: @cloudnautique I reordered the imports and updated PR

**Comment by cloudnautique** [comment]
Thanks, LGTM

**Comment by alena1108** [comment]
LGTM

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/3898

**Pull request merged**